src/astarte-client/client.ts

```diff
@@ -56,46 +57,44 @@ export interface AstarteInterfaceDescriptor {
   major: number;
   minor: number;
 }
-type Channel = any;
-type Trigger = any;
 
 type InterfaceOrInterfaceNameParams =
   | { interfaceName: AstarteInterface['name'] }
   | { interface: AstarteInterface };
 
 // Wrap phoenix lib calls in promise for async handling
 async function openNewSocketConnection(
-  connectionParams: any,
-  onErrorHanlder: any,
-  onCloseHandler: any,
-): Promise<PhoenixSocket> {
+  connectionParams: { socketUrl: string; realm: string; token: string },
+  onErrorHanlder: () => void,
+  onCloseHandler: () => void,
+): Promise<Socket> {
   const { socketUrl, realm, token } = connectionParams;
 
   return new Promise((resolve) => {
-    const phoenixSocket = new PhoenixSocket(socketUrl, {
+    const phoenixSocket = new Socket(socketUrl, {
       params: {
         realm,
         token,
       },
     });
-    phoenixSocket.onError((e: any) => onErrorHanlder(e));
-    phoenixSocket.onClose((e: any) => onCloseHandler(e));
+    phoenixSocket.onError(() => onErrorHanlder());
```

Comment:
this removes the ability to handle the event from the callback. Are we sure we want to do that? we lose the ability to understand why the error happened

Reply:
Since we are not using / parsing the error, I chose the simplest path of dropping the argument for now, instead of introducing another possibly heavy typing in this PR.
Of course, debatable

Reply:
the error comes from phoenix. it's probably exported by the phoenix library we're using

Reply:
That was my first assumption too, however I didn't find any useful reference and onError is typed as `Socket.onError(callback: (cb: any) => void): string`
So we're stuck with a `any` default type

Reply:
I simplified to just
```
phoenixSocket.onError(onErrorHanlder);
phoenixSocket.onClose(onCloseHandler);
```
That should be fine for now